content/spin/plugin-authoring.md

```diff
@@ -0,0 +1,41 @@
+title = "Creating Spin plugins"
+template = "spin_main"
+date = "2023-01-26T00:22:56Z"
+[extra]
+url = "https://github.com/fermyon/spin/blob/main/docs/content/plugin-authoring.md"
+---
+
+Spin plugins add new functionality or subcommands to Spin without modifying the Spin codebase. They make Spin easily extensible while keeping it lightweight. Spin plugins can add new triggers to Spin (such as the [example timer trigger](https://github.com/fermyon/spin/blob/main/examples/spin-timer/trigger-timer.json)), enable new language SDKs (such as [`js2wasm`](https://github.com/fermyon/spin-plugins/blob/main/manifests/js2wasm/js2wasm.json)), and more.
+
+This document will cover what Spin plugins are, how to use a plugin, and how to contribute a plugin.
+
+## What are Spin plugins?
+
+A Spin plugin is an executable that is added to Spin's plugins directory (`$XDG_DATA_HOME/spin/plugins`) upon a `spin plugins install <plugin-name>`. The plugin is then ready to be used. If the plugin is an extension to the Spin CLI, it can now be executed directly as a subcommand: `spin <plugin-name>`. If the plugin is a trigger plugin, it will be executed during `spin up` when an app using that trigger is run. 
+
+While for now plugins are assumed to be executables, in the future, support for plugging in WebAssembly modules may be desirable.
+
+## How to find and use a Spin plugin
+
+Spin maintains a [Spin plugins repository](https://github.com/fermyon/spin-plugins) that acts as an inventory of available plugins. The `spin plugins` command can be used to list plugins (available and installed) and install a new plugin. For example, the `js2wasm` plugin is needed in order to write JavaScript Spin applications. It can be installed from the Spin plugins repository, using the following command: 
+
+```sh
+spin plugins install js2wasm
+```
+
+During `spin build`, `spin js2wasm` is called to build the JS Spin application. Learn more about building Spin components in JavaScript [here](https://developer.fermyon.com/spin/javascript-components.md).
+
+## Authoring a Spin plugin
+
+As mentioned previously, Spin plugins are executables that are executed either directly via the Spin CLI or as a trigger. 
+
+Creating and using a plugin can be summarized as follows:
+1. Build, package, and zip the plugin
+1. Create a Spin plugin manifest for the plugin
```

Comment:
```suggestion
2. Create a Spin plugin manifest for the plugin
```

Reply:
Just to share context, in the past, i have found it a best practice to make numbered bullets in markdown always 1. and let markdown resolve the number ordering, making it easier to add items in the list in the future. Switching to manual ordering for now, though

Reply:
Good point, I did not realize the rendering worked with bullets always 1. Brilliant.